pkg/controller/perconaservermongodbbackup/perconaservermongodbbackup_controller.go

```diff
@@ -172,6 +177,13 @@ func (r *ReconcilePerconaServerMongoDBBackup) Reconcile(ctx context.Context, req
 		cluster = nil
 	}
 
+	if err := checkStartingDeadline(ctx, cluster, cr); err != nil {
+		if err := r.setFailedStatus(ctx, cr, err); err != nil {
+			return rr, errors.Wrap(err, "update status")
+		}
+		return reconcile.Result{}, nil
+	}
```

Comment:
By using the global `err` it's possible to set an error in the defer block without using the `setFailedStatus` method

```suggestion
	err = checkStartingDeadline(ctx, cluster, cr)
	if err != nil {
		return reconcile.Result{}, err
	}
```